onnxruntime/core/providers/cuda/tensor/nonzero_op.cc

```diff
@@ -11,10 +11,18 @@ namespace cuda {
 
 // kernel builder functions
 #define NONZERO_TYPED_KERNEL_WITH_TYPE_NAME(type, type_name)                       \
+  ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(                                         \
+      NonZero,                                                                     \
+      kOnnxDomain,                                                                 \
+      9, 12,                                                                       \
+      type_name,                                                                   \
+      kCudaExecutionProvider,                                                      \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<type>()), \
+      NonZero<type>)                                                               \
   ONNX_OPERATOR_TYPED_KERNEL_EX(                                                   \
       NonZero,                                                                     \
       kOnnxDomain,                                                                 \
-      9,                                                                           \
```

Comment:
>9 [](start = 6, length = 1)

Since there is no kernel change, 9~13 are still effectively sharing the same mapping to kernel... 

Is the declaration of opset 13 kernel absolutely needed? for new opset 13 op, can it fall back to the previous version of the kernel? 

This could potentially save a lot of changes, but I am not sure if it's feasible to do so.

Reply:
Cheng mentioned in one of his PR that our kernel registry doesn't support such fall back. He already declared opset 13 nernels for some of the CPU ops, I think it's better we jsut follow such rule to make it consistent.